src/Agoda.Analyzers/AgodaCustom/AG0042QuerySelectorShouldNotBeUsed.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Immutable;
+using Agoda.Analyzers.Helpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Agoda.Analyzers.AgodaCustom
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class AG0042QuerySelectorShouldNotBeUsed : DiagnosticAnalyzer
+    {
+        public const string DIAGNOSTIC_ID = "AG0042";
+
+        private static readonly LocalizableString Title = new LocalizableResourceString(
+            nameof(CustomRulesResources.AG0042Title), CustomRulesResources.ResourceManager,
+            typeof(CustomRulesResources));
+
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(
+            nameof(CustomRulesResources.AG0042Title), CustomRulesResources.ResourceManager,
+            typeof(CustomRulesResources));
+
+        private static readonly LocalizableString Description
+            = DescriptionContentLoader.GetAnalyzerDescription(nameof(AG0042QuerySelectorShouldNotBeUsed));
+
+        public static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+            DIAGNOSTIC_ID,
+            Title,
+            MessageFormat,
+            AnalyzerCategory.CustomQualityRules,
+            DiagnosticSeverity.Warning,
+            AnalyzerConstants.EnabledByDefault,
+            Description,
+            "https://playwright.dev/dotnet/docs/api/class-elementhandle",
+            WellKnownDiagnosticTags.EditAndContinue);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(AnalyzeSyntaxNode, SyntaxKind.InvocationExpression);
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
+
+        private static void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpression = (InvocationExpressionSyntax)context.Node;
+
+            var memberAccess = invocationExpression.Expression as MemberAccessExpressionSyntax;
+            if (memberAccess == null)
+                return;
+
+            var methodName = memberAccess.Name.Identifier.Text;
+
+            // Check if the method is QuerySelectorAsync
```

Comment:
Do these comments really add value? I mean, do we really need to explain what `if (methodName != "QuerySelectorAsync")` means?

Reply:
Removed those already, yeah.
it was just copy-paste from GPT, my bad that I didn't clean it up :D 
Thanks for catching it!